blockchain/addressGenerator_test.go

```diff
@@ -1,69 +1,163 @@
 package blockchain
 
 import (
-	"crypto/rand"
 	"fmt"
 	"testing"
 
+	"github.com/ElrondNetwork/elrond-go-core/core/mock"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
 	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestAddress_GetShard(t *testing.T) {
+func TestNewAddressGenerator(t *testing.T) {
 	t.Parallel()
 
-	pubkey := make([]byte, 32)
-	_, _ = rand.Read(pubkey)
-
-	numShardsWithoutMeta := uint32(2)
-	shardCoordinator, _ := NewShardCoordinator(numShardsWithoutMeta, 0)
-
-	pubkey[31] &= 0xFE
-	addr0 := data.NewAddressFromBytes(pubkey)
-
-	pubkey[31] |= 0x01
-	addr1 := data.NewAddressFromBytes(pubkey)
-
-	sh0, err := shardCoordinator.ComputeShardId(addr0)
-	assert.Nil(t, err)
-
-	sh1, err := shardCoordinator.ComputeShardId(addr1)
-	assert.Nil(t, err)
-
-	assert.Equal(t, sh0, uint32(0))
-	assert.Equal(t, sh1, uint32(1))
+	t.Run("nil pubkey converter", func(t *testing.T) {
+		t.Parallel()
+
+		args := ArgsAddressGenerator{
+			PubkeyConv: nil,
+		}
+		ag, err := NewAddressGenerator(args)
+
+		assert.Nil(t, ag)
+		assert.Equal(t, process.ErrNilPubkeyConverter, err)
+	})
+	t.Run("nil address generator core", func(t *testing.T) {
+		t.Parallel()
+
+		args := ArgsAddressGenerator{
```

Comment:
can you explicitly set the AddressGeneratorCore to Nil?

Reply:
done